src/playwrightTest.ts

```diff
@@ -222,6 +229,52 @@ export class PlaywrightTest {
     await reporterServer.wireTestListener(listener, token);
   }
 
+  async runBrowserServer(workspaceFolder: string, token: vscodeTypes.CancellationToken) {
+    if (this._browserServerWS)
+      return;
+
+    const npx = await this.findNpx();
+
+    const allArgs = [
+      'playwright',
+      'run-server',
+      '--reuse-browser',
+      `--path=/${createGuid()}`
+    ];
+
+    const serverProcess = spawn(npx, allArgs, {
+      cwd: workspaceFolder,
+      stdio: ['pipe', 'pipe', 'pipe'],
+    });
+
+    const stdio = serverProcess.stdio;
+    stdio[1].on('data', data => {
+      const match = data.toString().match(/Listening on (.*)/);
+      if (!match)
+        return;
+      this._browserServerWS = match[1];
+    });
+
+    stdio[2].on('data', data => {
+      console.log('Error in browser server: ' + data.toString());
+    });
+
+    token.onCancellationRequested(() => {
+      serverProcess.stdin.write('<EOL>');
```

Comment:
I think we used to `stdin.close()` in other places instead? That would just imitate disconnect.

Reply:
Unfortunately, it does not. I think we were doing it in our Java port only.